src/components/Context/CartContext.tsx

```diff
@@ -0,0 +1,100 @@
+import { PropsWithChildren, createContext, useState } from "react";
+import { FeaturedProductObject } from "../App";
+
+export interface ItemToPurchase {
+  quantity: number;
+  product: FeaturedProductObject;
+}
+
+export const CartContext = createContext<ItemToPurchase[]>([]);
+
+export const CartProvider = ({ children }: PropsWithChildren) => {
+  const [itemsInCart, setItemsInCart] = useState<ItemToPurchase[]>([]);
+
+  const numItemsInCart = () => {
+    return itemsInCart.length;
+  };
+
+  const addItemToCart = (itemToPurchase: ItemToPurchase) => {
+    const isItemInCart = itemsInCart.find((itemInCart) => {
```

Comment:
Why some instead of find here?  User should only be adding one item anytime this function runs, even if they are adding multiple quantities of that item.  .some() returns true if any of the items in an array match a condition, right?  Could this be because we only need a true or false return?  I know we already have access to the itemToPurchase as it is a parameter to addItemToCart

Reply:
> Could this be because we only need a true or false return

yep exactly! `some` is a slightly better fit for the job of determining whether some item in the cart has the given id, and based on the variable name `isItemInCart`, it should be a boolean instead of an item